### PR TITLE
fix(badge): dpo tokens

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. use dpo status tokens in badge
2. deprecate old statuses and alias them to dpo statuses

### CODE REVIEW

File: elements/rh-badge/rh-badge.ts [hunk 1]
```diff
@@ -28,12 +30,17 @@ export class RhBadge extends LitElement {
...
+    | 'caution'
+    | 'neutral'
     | 'info'
     | 'success'
-    | 'moderate'
-    | 'important'
-    | 'critical';
+    | 'moderate' // deprecated
+    | 'important' // deprecated
+    | 'critical' = // deprecated
```
Comment (adamjohnson) [inline elements/rh-badge/rh-badge.ts:42]: `critical` is not showing as deprecated in [the docs](https://deploy-preview-1912--red-hat-design-system.netlify.app/elements/badge/code/#rh-badge) (assuredly due to the `=`).

Resize your browser to ~1100px to make this point clear.

**Review by marionnegp — COMMENTED** [review]
@coreyvickery, looking at the [status colors file](https://www.figma.com/design/FAhXoct6VdXsoRDkwHBLMK/Status-colors?node-id=193-525&m=dev) vs this PR's [color context demo](https://deploy-preview-1912--red-hat-design-system.netlify.app/elements/badge/demo/color-context/), there are some `<rh-badge>` colors that don't line up with the status tokens we have:
- Under light theme: The yellow`warning` and gray `neutral` badges in the Figma file use colors that aren't currently status tokens.
- Under dark theme: The gray `neutral` badge doesn't use a dark theme status token.

What do you think about changing `<rh-badge>` in Figma to look the badges in the demo? It would keep the use of colors more consistent and make theming easier. If we go with what's in the Figma file, we'll either have to use crayon colors or creating new status tokens.

**Review by adamjohnson — CHANGES_REQUESTED** [review]
**`20-guidelines.md`:**

The table under the "Variants" heading still lists deprecated states:

  * "Moderate" changes to "Caution"
  * "Important" changes to "Warning"
  * "Critical" changes to "Danger"

Screenshot of the badge statuses table on the Guidelines page in the docs

**Comment by coreyvickery** [comment]
@marionnegp Sounds good, do you want to handle that?

**Review by markcaron — APPROVED** [review]
Logical Goal: Tokens Mapping

**Comment by marionnegp** [comment]
@coreyvickery, yep! I can update the Figma component by tomorrow. 

@bennypowers, I'll work on updating the "Variants" section of the docs today. I'll go ahead and use this same branch.

File: elements/rh-badge/rh-badge.ts (hunk 1 above)
Comment (markcaron) [reply elements/rh-badge/rh-badge.ts:42]: @adamjohnson I could be wrong, but I believe that `=` is the assignment of `neutral` should it be `null`.

Comment (markcaron) [reply elements/rh-badge/rh-badge.ts:42]: @adamjohnson OH, I get it. I'll see if I can fix that.

File: .changeset/tall-bags-pipe.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+"@rhds/elements": minor
+---
+`<rh-badge>`: added new states: `danger`, `warning`, `caution`, `neutral`, and 
+`caution`, and deprecated `moderate` (now an alias to `caution`), `important`
```
[the remark this answers is not in the document]
Comment (marionnegp) [reply .changeset/tall-bags-pipe.md]: I think `moderate` should alias to `warning`, and `important` should alias to `caution`.

File: elements/rh-badge/rh-badge.ts (hunk 1 above)
Comment (bennypowers) [reply elements/rh-badge/rh-badge.ts]: The rhs after the = is the initializer

Comment (adamjohnson) [reply elements/rh-badge/rh-badge.ts]: To please the docs parser, I think we're going to have to do this:

```ts
  /**
   * Denotes the state-of-affairs this badge represents
   */
  @property({ reflect: true }) state:
    | 'danger'
    | 'warning'
    | 'caution'
    | 'neutral'
    | 'success'
    | 'moderate' // deprecated
    | 'important' // deprecated
    | 'critical' // deprecated
    | 'info' =
      'neutral';
```

Comment (bennypowers) [reply elements/rh-badge/rh-badge.ts]: Or move the inline comments to the jsdoc

File: .changeset/tall-bags-pipe.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamjohnson) [reply .changeset/tall-bags-pipe.md]: Updated the changeset in [0fa0d5e](https://github.com/RedHat-UX/red-hat-design-system/pull/1912/commits/0fa0d5eecc6761cbe014ed919835acea7b7493c7). 

<hr>

@marionnegp I updated `moderate/warning` and `important/caution` in [6765a92](https://github.com/RedHat-UX/red-hat-design-system/pull/1912/commits/6765a92247cdec627408cc1b49ac3266231ef9a7). Can you verify each badge looks as it should?

Also, while it passes WCAG color contrast guidelines, the [caution badge](https://deploy-preview-1912--red-hat-design-system.netlify.app/elements/badge/demo/caution/) text color is very dark. We're explicitly making it `--rh-color-text-primary-on-light, #151515` for `.on.light.warning` and `on.light.caution` in `rh-badge.css`.

Does that look correct?

FWIW, if we remove that CSS rule, it still passes WCAG color contrast guidelines (and has white text).

File: elements/rh-badge/rh-badge.ts (hunk 1 above)
Comment (adamjohnson) [reply elements/rh-badge/rh-badge.ts]: Resolved by [7defe3b](https://github.com/RedHat-UX/red-hat-design-system/pull/1912/commits/7defe3b102061992cfc2057884527676f8fba9b7).

File: .changeset/tall-bags-pipe.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (marionnegp) [reply .changeset/tall-bags-pipe.md]: Yep, those changes look correct! 

For the caution badge, I'm getting a contrast of 3.7:1 for white on orange-50. 😬

**Comment by marionnegp** [comment]
I've updated images in the docs and added back a length token for a min width of 32px, which includes padding.

File: .changeset/tall-bags-pipe.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamjohnson) [reply .changeset/tall-bags-pipe.md]: Right you are. Better color contrast in its current state (not just AA Large for white text). Maybe not for actual humans though. 🫤

I can't wait until we get to use the [APCA](https://www.linkedin.com/pulse/understanding-apca-wcag-31-new-era-color-contrast-digital-boiano-itmyf/) to determine color contrast.

Thanks for the confirmation.

**Review by marionnegp — APPROVED** [review]
Element and changeset info looks good!

**Pull request merged**